### PR TITLE
Add Docker HealthCheck and linux package to support container healthcheck

### PR DESCRIPTION
**Pull request by rafaelherik**
## What does this PR do?

Solve #417

Add `procps` and `net-tools` to the Docker Image to support infrastructure health checks. It doesn't implement the application HealthCheck endpoint, but gives the possibility to check if the server is running correctly and exposed to a TCP Port.

 Add `HEALTHCHECK` to the Dockerfile. 

## Checklist before merging
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-mcp/blob/main/CONTRIBUTING.md) on pull request process, code style, and testing.**
- [x] Title of the pull request is clear and informative.
- [x ] There are a small number of commits, each of which have an informative message.  This means that previously merged commits do not appear in the history of the PR.  For more information on cleaning up the commits in your PR,  [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] For core features, I have added thorough tests.
- [ ] For user-impacting changes (bug fixes, new features, UI/UX changes), I have added a CHANGELOG.md entry linking to this PR.
- [ ] Have a team member run [live tests](https://github.com/Azure/azure-mcp/blob/main/CONTRIBUTING.md#live-tests):
   - [ ] Team Member: Inspect PR for security issues before queueing a test run
   - [ ] Team Member: Add this comment to the pr `/azp run azure - mcp` to start the run

### CODE REVIEW

**Comment by rafaelherik** [comment]
@microsoft-github-policy-service agree

File: Dockerfile [hunk 1]
```diff
@@ -30,10 +30,15 @@ RUN apt-get update && apt-get install -y \
...
     gnupg \
+    net-tools \
+    procps \
     && curl -sL https://aka.ms/InstallAzureCLIDeb | bash \
     && rm -rf /var/lib/apt/lists/*
 
 # Copy the published application
 COPY --from=publish "/app/publish/linux-x64/dist" .
 
+HEALTHCHECK --interval=30s --timeout=10s --start-period=5s --retries=3 \
```
Comment (jongio) [inline Dockerfile:41]: Are there any tests we can write to verify this is working?

Comment (rafaelherik) [reply Dockerfile]: @jongio good catch, your suggestion prompted me to write test scripts (`test-healthcheck.sh`and `test-healthcheck.ps1`) - I am more familiar with bash, so I kept it and added also pwsh  to validate the health check functionality. During test development, I discovered some edge cases that refined the implementation:

### testing challenges:

**False positive testing:** Hard to simulate "unhealthy but running" scenarios since containers exit when azmcp stops
Kubernetes timing gap: Without health checks, containers appear "ready" before the service endpoint is actually available, causing traffic routing issues

### what the tests verify:
- Container starts with health check configured
- Health check script executes correctly
- SSE mode exposes the correct port
- Health checks are non-intrusive (no HTTP requests to main service)
- STDIO mode behaves as expected

I also added `docs/container-usage.md` with Kubernetes deployment guidance and improved error handling in the test scripts. 

The tests demonstrate that proper health checks solve the timing issue where pods receive traffic before they're ready to serve requests.

Comment (jongio) [reply Dockerfile]: Is there any way to automate these tests so we can run in CI?

**Comment by jongio** [comment]
Sorry for the inconvenience, but we have moved the Azure MCP Server source code to https://github.com/microsoft/mcp.  This change allows us to build any Microsoft MCP server with the same engineering system and allows us to brand any MCP server with either Azure or Microsoft branding.  This repo change means that you need to move this PR to that new repo.  Please do so and then add a link here to that new PR.  We'll close this PR now and we look forward to seeing this over on the new repo.
